etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/ruleengine/RuleEngine.java

```diff
@@ -28,7 +28,7 @@ public void unloadRules() {
         rules.clear();
     }
 
-    public void ensureRulesLoaded() {
+    public synchronized void ensureRulesLoaded() {
```

Comment:
This is in a performance sensitive flow during API flows, and synchronizing here will slow down concurrent calls to our API.

We can prevent this bottleneck by doing "double-checked locking".

```java
public void ensureRulesLoaded() {
    if (rules.isEmpty()) {
        synchronized(this) {
            if (rules.isEmpty()) {
                loadRules();
            }
        }
    }
}
```

This also allows us to remove `synchronized` from the `loadRules` method below.

Reply:
That's a really good catch! Updating...